doozer/doozerlib/cli/release_gen_payload.py

```diff
@@ -603,15 +607,37 @@ def generate_payload_entries(self, assembly_inspector: AssemblyInspector) -> (Di
                 continue
             # No adjustment for private or public; the assembly's canonical payload content is the same.
 
-            entries: Dict[str, PayloadEntry]  # Key of this dict is release payload tag name
-            issues: List[AssemblyIssue]
-            entries, issues = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PUBLIC))
+            # public_entries: Dict[str, PayloadEntry]  # Key of this dict is release payload tag name
+            # public_payload_issues: List[AssemblyIssue]
+            entries, payload_issues = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PUBLIC))
+
+            public_entries: Dict[str, PayloadEntry] = dict()
+            for k, v in entries.items():
+                if not v.build_inspector:
+                    # Its RHCOS, since it doesn't have a build inspector. Put it in for now, but change once RHCOS
+                    # supports private nightlies
+                    public_entries[k] = v
+                    continue
+
+                if v.build_inspector.is_under_embargo() and self.runtime.assembly_type == AssemblyTypes.STREAM:
+                    # It's an embargoed build. Filter it out if its stream
+                    continue
+
+                public_entries[k] = v
 
-            # Exclude issues for now, for private repo
-            private_entries, _ = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PRIVATE))
             entries_for_arch[arch] = entries
```

Comment:
This should be public_entries i think since entries_for_arch is how the payload will be constructed
```suggestion
            entries_for_arch[arch] = public_entries
```